src/perm.c

```diff
@@ -9,85 +9,26 @@
 /// \param ordinal The ordinal as the input.
 /// \param mismatches How many bits to set.
 /// \param subkey_length How big the actual bit string is in bits
-void decode_ordinal(mpz_t perm, const mpz_t ordinal, int mismatches, size_t subkey_length) {
-    mpz_t binom, curr_ordinal;
-    mpz_inits(binom, curr_ordinal, NULL);
-
-    mpz_set(curr_ordinal, ordinal);
-
-    mpz_set_ui(perm, 0);
-    for (unsigned long bit = subkey_length - 1; mismatches > 0; bit--) {
-        mpz_bin_uiui(binom, bit, mismatches);
-        if (mpz_cmp(curr_ordinal, binom) >= 0) {
-            mpz_sub(curr_ordinal, curr_ordinal, binom);
-            mpz_setbit(perm, bit);
-            mismatches--;
-        }
-    }
-
-    mpz_clears(binom, curr_ordinal, NULL);
-}
-
-void get_random_permutation(mpz_t perm, int mismatches, size_t subkey_length,
-                            gmp_randstate_t randstate) {
-    mpz_t ordinal, binom;
-    mpz_inits(ordinal, binom, NULL);
-
-    mpz_bin_uiui(binom, subkey_length, mismatches);
-
-    mpz_urandomm(ordinal, randstate, binom);
-    decode_ordinal(perm, ordinal, mismatches, subkey_length);
-
-    mpz_clears(ordinal, binom, NULL);
-}
-
-void get_benchmark_permutation(mpz_t perm, int mismatches, size_t subkey_length,
-                               gmp_randstate_t randstate, int numcores) {
-    mpz_t ordinal, binom, rank, cores;
-    mpz_inits(ordinal, binom, rank, NULL);
-    mpz_init_set_ui(cores, numcores);
-
-    mpz_bin_uiui(binom, subkey_length, mismatches);
-
-    // Choose a random rank from 0 to numcores - 1
-    mpz_urandomm(rank, randstate, cores);
-
-    mpz_mul_ui(rank, rank, 2);
-    mpz_add_ui(rank, rank, 1);
-    mpz_mul(ordinal, binom, rank);
-
-    mpz_tdiv_q_ui(ordinal, ordinal, numcores * 2);
-
-    decode_ordinal(perm, ordinal, mismatches, subkey_length);
-
-    mpz_clears(ordinal, binom, rank, cores, NULL);
-}
+void decodeOrdinal(mpz_t perm, const mpz_t ordinal, int mismatches, size_t subkey_length);
```

Comment:
Whered the definition go?

Reply:
Over to line 97 (I separated declarations and definitions similar to me responding down below about public before private)